pkg/controller/core/workload_controller.go

```diff
@@ -154,9 +181,57 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		err := workload.ApplyAdmissionStatus(ctx, r.client, &wl, true)
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
+
 	return ctrl.Result{}, nil
 }
 
+func (r *WorkloadReconciler) getClusterQueueAdmissionChecks(ctx context.Context, name string) ([]string, error) {
+	queue := kueue.ClusterQueue{}
+	if err := r.client.Get(ctx, types.NamespacedName{Name: name}, &queue); err != nil {
+		return nil, err
+	}
+	return queue.Spec.AdmissionChecks, nil
+}
+
+func syncAdmissionCheckConditions(conds []metav1.Condition, queueChecks []string) ([]metav1.Condition, bool) {
+	shouldUpdate := false
+	if len(queueChecks) == 0 {
+		if len(conds) > 0 {
+			conds = nil
+			shouldUpdate = true
+		}
+	} else {
+		currentChecks := slices.ToRefMap(conds, func(c *metav1.Condition) string { return c.Type })
+		for _, t := range queueChecks {
+			if _, found := currentChecks[t]; !found {
+				apimeta.SetStatusCondition(&conds, metav1.Condition{
+					Type:   t,
+					Status: metav1.ConditionUnknown,
+					Reason: string(kueue.CheckStatePending),
+				})
+				shouldUpdate = true
+			}
+		}
+
+		// if the workload conditions length is bigger then some cleanup should be done
+		if len(conds) > len(queueChecks) {
+			toRemove := make([]string, 0, len(conds)-len(queueChecks))
+			queueChecksSet := sets.New(queueChecks...)
+			for i := range conds {
+				c := &conds[i]
+				if !queueChecksSet.Has(c.Type) {
+					toRemove = append(toRemove, c.Type)
+				}
+			}
+			shouldUpdate = len(toRemove) > 0
+			for _, t := range toRemove {
+				apimeta.RemoveStatusCondition(&conds, t)
+			}
```

Comment:
A simpler and more performant implementation is to build a new slice of conditions: the conditions that we need to keep.

Reply:
not more performant... however done